src/cls/rbd/cls_rbd_types.h

```diff
@@ -531,18 +531,38 @@ inline void decode(TrashImageSource &source, bufferlist::const_iterator& it)
   source = static_cast<TrashImageSource>(int_source);
 }
 
+enum TrashImageState {
+  TRASH_IMAGE_STATE_NORMAL = 0,
+  TRASH_IMAGE_STATE_DELETING = 1,
```

Comment:
Nit: perhaps just name it ```TRASH_IMAGE_STATE_PENDING``` so that you can re-use the state when adding an image to the trash (and set it to normal when it's fully moved into the trash).